src/wwwroot/app/directives/candidate-vacancy-info/candidate-vacancy-info.js

```diff
@@ -29,179 +35,229 @@ export default class CandidateVacancyInfoDirective {
+               case STAGE_STATES.Passed: {
+                  stageState = STAGE_STATES.Passed;
+                  vsiClass = 'passed';
+                  break;
+               }
+               case STAGE_STATES.NotPassed: {
+                  stageState = STAGE_STATES.NotPassed;
+                  vsiClass = 'not-passed';
+                  break;
+               }
+               default: {
+                  stageState = STAGE_STATES.Inactive;
+                  vsiClass = 'inactive';
+                  break;
+               }
+            }
+            if (stageVacancyStageInfo.stage.isCommentRequired) {
+               showCommentArea = true;
+            }
          }
-      } else {
-         _addVSI(stagesObject, stageToPass, comment, isPassed);
-      }
-      deffer.resolve(stagesObject, stageToPass);
-      return deffer.promise;
+         return {
+            stage: vacancyStage,
+            vsi: stageVacancyStageInfo,
+            showCommentArea,
+            stageState,
+            class: vsiClass
+         };
+      }));
    }
 
-   vm.getCommentByStageId = (stages, stageId, candidateId) => {
-      stageId = parseInt(stageId);
-      let comment = '';
-      let vacancyStageInfo = find(stages, { candidateId, stageId });
-      if (vacancyStageInfo) {
-         if (vacancyStageInfo.comment) {
-            comment = vacancyStageInfo.comment.message;
+   function updateCandidateStagesForWith(entityStageObject, vacancyStagesEntitiesVSIs) {
+      entityStageObject.vacancyStageInfos = map(filter(vacancyStagesEntitiesVSIs, (stageVsi) => {
+         if (stageVsi.vsi) {
+            return true;
          }
-      }
-      return comment;
-   };
-
-   vm.passStageToNext = (candidateStagesObject) => {
-      candidateStagesObject.currentStageId = candidateStagesObject.selectedStageId;
-      candidateStagesObject.selectedStageId = candidateStagesObject.selectedStageId + 1;
-      changeStage(candidateStagesObject).then(() => {
-         _handleStageChanged(candidateStagesObject, candidateStagesObject.selectedStageId);
-      }).catch((currentStage) => {
-         notifySelectedStageChangingBack();
-         candidateStagesObject.selectedStageId = currentStage.id;
+      }), (stageVsi) => {
+         return stageVsi.vsi;
       });
-   };
+      recalculateCurrentStageId(entityStageObject);
+      vm.stagesToShow = calculateVacancyStagesEntitiesCount(vm.vacancyStageInfosComposedByCandidateIdVacancyId);
+      return $q.when();
+   }
 
-   vm.onStageChange = (stagesObject) => {
-      changeStage(stagesObject).then(() => {
-         _handleStageChanged(stagesObject, stagesObject.selectedStageId);
-      }).catch((currentStage) => {
-         notifySelectedStageChangingBack();
-         stagesObject.selectedStageId = currentStage.id;
+   function keepRejectVSIs(entityStageObject, vacancyStages) {
+      let rejectStages = filter(vacancyStages, (extStage) => {
+         return extStage.stage.stageType === STAGE_TYPES.RejectStage;
```

Comment:
you don't need figure braces,
your code can be next:
let rejectStages = filter(vacancyStages, extStage => extStage.stage.stageType === STAGE_TYPES.RejectStage);